internal/checker/utilities.go

```diff
@@ -1218,8 +1218,84 @@ func isValidNumberString(s string, roundTripOnly bool) bool {
 	return !n.IsNaN() && !n.IsInf() && (!roundTripOnly || n.String() == s)
 }
 
-func isValidBigIntString(s string, roundTripOnly bool) bool {
-	return false // !!!
+func isValidBigIntString(s string, roundtrip bool) bool {
+	if s == "" {
+		return false
+	}
+
+	// Create a scanner and set the text to the input string with an 'n' suffix
+	scan := scanner.NewScanner("", scanner.ScriptTargetESNext, false /*skipTrivia*/)
+	text := s + "n"
+	scan.SetText(text)
```

Comment:
It seems like this code is missing the `onError` stuff from the original function.

Reply:
updated